auth/src/main/java/com/firebase/ui/auth/util/SignInDelegate.java

```diff
@@ -0,0 +1,450 @@
+                    // Smart lock selector cancelled, go to the AuthMethodPicker screen
+                    startAuthMethodChoice();
+                } else if (resultCode == RESULT_FIRST_USER) {
+                    // TODO: (serikb) figure out flow
+                }
+                break;
+            case RC_IDP_SIGNIN:
+            case RC_AUTH_METHOD_PICKER:
+            case RC_EMAIL_FLOW:
+                mAuthUIResult.onResult(resultCode, new Intent());
+                break;
+            case RC_PLAY_SERVICES:
+                if (resultCode != RESULT_OK) {
+                    mAuthUIResult.onResult(resultCode, new Intent());
+                }
+                break;
+        }
+    }
+
+    private String getEmailFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getId();
+    }
+
+    private String getAccountTypeFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getAccountType();
+    }
+
+    private String getPasswordFromCredential() {
+        if (mCredential == null) {
+            return null;
+        }
+        return mCredential.getPassword();
+    }
+
+    private void handleCredential(Credential credential) {
+        mCredential = credential;
+
+        if (IdentityProviders.GOOGLE.equals(credential.getAccountType())) {
+            // Google account, rebuild GoogleApiClient to set account name and then try
+            initGoogleApiClient(credential.getId());
+            googleSilentSignIn();
+        } else {
+            // Email/password account
+            String status = String.format("Signed in as %s", credential.getId());
+            Log.d(TAG, status);
+        }
+    }
+
+    private void initGoogleApiClient(String accountName) {
+        GoogleSignInOptions.Builder gsoBuilder = new GoogleSignInOptions
+                .Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestEmail();
+
+        if (accountName != null) {
+            gsoBuilder.setAccountName(accountName);
+        }
+
+        if (mGoogleApiClient != null) {
+            mGoogleApiClient.disconnect();
+        }
+
+        mGoogleApiClient = new GoogleApiClient.Builder(mActivity)
+                .addConnectionCallbacks(this)
+                .addApi(Auth.CREDENTIALS_API)
+                .addApi(Auth.GOOGLE_SIGN_IN_API, gsoBuilder.build())
+                .build();
+        mGoogleApiClient.connect();
+    }
+
+    private void resolveSavedEmails() {
+        if (mCredentialRequestResult == null || mCredentialRequestResult.getStatus() == null) {
+            return;
+        }
+        Status status = mCredentialRequestResult.getStatus();
+        if (status.getStatusCode() == CommonStatusCodes.RESOLUTION_REQUIRED) {
+            try {
+                // TODO check the mActivity stuff
+                status.startResolutionForResult(mActivity, RC_CREDENTIALS_READ);
```

Comment:
Here's where I'm stuck. There are two options: either people using the library would have to call `onActivityResult` for us, or I would have to change `SignInDelegate` to an `AppCompatFragment` which would mean people using the library have to pass in an `AppCompatActivity` instead of being able to pass in any activity. I'm in favor of the latter because 99.9% of people are probably already using `AppCompatActivity` anyway since it is necessary for the `Toolbar` (added in API v21).